Sources/JSON.swift

```diff
@@ -0,0 +1,50 @@
+public enum JSON {
+    public enum Number {
+        case integer(Int)
+        case unsignedInteger(UInt)
```

Comment:
Actually, as far as I can tell we don't have a way to discern between Int and UInt during parsing :/ I'd either just keep (Signed) Int and Double or move to @paulofaria's C7 Number type. Either one is fine with me.